### PR TITLE
feat(cli): warn when a resource pin loses its timer claim

### LINKED ISSUE #1412: chore(cli): add live-claim-status diagnostic for resource/timer (low-priority, EF-original)
AI Generated issue-ticket

## Summary

`resource`/`timer` has no live-claim-status diagnostic. A user configuring a `resource`+`timer` assignment that will lose an ownership collision (e.g. two roles pointed at the same physical timer pin) gets no CLI feedback that the config is dead on arrival — `resource show` only echoes back configured *intent* (whatever was `SET`), not whether `timerAllocate()` actually succeeded for it at boot.

## Precedent — this is not a new pattern, EF already built it once

`printDmaoptClaimStatus()` (IT#1388 / PR#1400, merged) does exactly this for the `dmaopt` CLI subsystem: it cross-checks a configured `dmaopt` value against live `dmaGetOwner()` state and warns on the `dma` get/list/set path when a saved config lost its claim to another peripheral.

## Confirmed: not a BF-porting gap, EF-original going forward

Checked BF 4.5-maintenance source directly — BF has no equivalent live-claim-status diagnostic for `dmaopt`, `resource`, or `timer`. `printDmaoptClaimStatus()` was an EF-original addition beyond BF's baseline, not a port. A `timerOwners[]`-equivalent version for `resource`/`timer` would be new EF-original work extending EF's own precedent, not something to look for or diff against BF going forward.

## Discovered

Surfaced 2026-08-25 while hardware-verifying emuflight/EmuFlight#1406 (`timerAllocate()` ownership-check port) on real `FOXEERF722V4`. Deliberately configured a `LED_STRIP`/`MOTOR` pin collision to test the ownership check — `resource show` displayed the losing `LED_STRIP` assignment as if it were live, and `dma` (not `resource`) was the only way to confirm it had actually been rejected at runtime.

## Scope, low-priority

- Not a bug in PR#1406's ownership-check logic itself — that logic is correct and hardware-verified.
- Evaluate-only / low-priority, matching the existing #1387/#1396-style items in this chain: a UX/diagnostics improvement, no known active flight-safety impact.
- Natural template: `printDmaoptClaimStatus()` (`interface/cli.c`, spawned by PR#1400) — same cross-check shape (configured value vs. live ownership accessor), applied to `timerOwners[]`/`timerGetOwner()`-equivalent state instead of `dmaGetOwner()`.

## Related

- emuflight/EmuFlight#1406 — where this was found
- emuflight/EmuFlight#1388 / PR#1400 — the `dmaopt` precedent this would extend

### PR DESCRIPTION
**Pull request by nerdCopter**
AI Generated pull-request

## Summary
- `resource`/PG state shows configured intent, not whether `timerAllocate()` actually won the pin's timer claim at boot -- a losing driver returns before ever calling `IOInit()`, so `resource show` still displays the losing assignment as if it were live.
- Adds `timerGetOwner()` (`timer_common.c`) to read `timerOwners[]`'s live per-pin state, and `printResourceClaimStatus()`/`printResourceClaimStatusAll()` (`cli.c`) to cross-check it against `resourceTable`, mirroring `printDmaoptClaimStatus()` (#1400).
- Wired into `resource` (bare) and after a `resource <owner> <index> <pin>` set; kept out of `printResource()`/config dump, matching the dmaopt precedent.
- EF-original diagnostic, not a BF port -- BF has no equivalent for `dmaopt`, `resource`, or `timer`.

Closes #1412.

## Test plan
- [x] `make clean_test && make test`: 49/49 test binaries pass, including 4 new `ResourceClaimStatusTest` cases in `cli_unittest.cc`
- [x] Build passes: HELIOSPRING, TUNERCF405, SKYSTARSF405AIO, PYRODRONEF7, FOXEERF722V4, FOXEERF405, APEXF7, TMOTORF7, STM32F7X2, NERO, SITL (11/11)
- [x] Compile-only: MATEKF411, NBDHMBF4PRO, WORMFC, ALIENWHOOPF7, CRAZYFLIE2 (5/5)
- [ ] Hardware verified: n/a -- CLI text-output diagnostic only, no runtime allocation/driver behavior change

## Summary by CodeRabbit

* **New Features**
  * Added resource ownership visibility for timer-managed pins.
  * The resource CLI now reports when a pin’s configured owner differs from its active timer owner.
  * Ownership status is shown both for individual assignments and when listing all resources.

* **Bug Fixes**
  * Improved detection and reporting of conflicting resource claims.

* **Tests**
  * Added coverage for matching, free, unassigned, and conflicting timer ownership states.

### CODE REVIEW

**Comment by nerdCopter** [comment]
@CodeRabbitAI — AI generated comment

Analysis requested on specific points in this diff:

1. `printResourceClaimStatus()` (cli.c) compares `timerGetOwner(ioTag)` only by owner *type*, not per-instance index (timerOwners[] never tracked resourceIndex — see the UNUSED(resourceIndex) comment in timerAllocate()). Confirm this can't produce a misleading false-negative: e.g. resourceTable configures MOTOR index 3 for a pin, but MOTOR index 5 actually won the timer claim on that same physical timer channel via a different ioTag mapping — would the check silently treat that as "own claim, no warning" when it's actually a different motor instance holding it?

2. `printResourceClaimStatus()`/`printResourceClaimStatusAll()` are deliberately NOT called from `printResource()` (which the general `dump`/`diff` config-output path also calls), matching the precedent set by `printDmaoptClaimStatus()` (never called from the dump path either). Confirm this reasoning holds and there's no other code path where `printResource()` is invoked in a context where the diagnostic actually should fire but currently doesn't.

3. `printResourceClaimStatusAll()` calls `getIoTag()` for every resourceTable entry unconditionally; `getIoTag()` dereferences `pgFind(value.pgn)` without a NULL check (pre-existing pattern, also used by `resourceCheck()`). Confirm `pgFind()` can never return NULL for any entry actually compiled into `resourceTable[]` (i.e. every PG referenced there is always registered), or flag if there's a realistic path where it could.

4. `timerGetOwner()` in timer_common.c returns `OWNER_FREE` both when a pin has no runtime timer-pinmap entry at all AND when `timerAllocate()` was simply never called yet this boot for an otherwise-valid pinmap entry. Confirm this ambiguity is acceptable for a low-priority UX diagnostic (i.e. it can't mask a real, currently-live conflict) — no case in the driver init order where a timer-consuming driver could be conditionally skipped this boot in a way that would hide a real conflict from the user.

5. General: any correctness or edge-case issues in the diff as a whole (`src/main/drivers/timer.h`, `src/main/drivers/timer_common.c`, `src/main/interface/cli.c`, `src/test/unit/cli_unittest.cc`).

File: src/main/interface/cli.c [hunk 1]
```diff
@@ -3894,6 +3894,32 @@ static void resourceCheck(uint8_t resourceIndex, uint8_t index, ioTag_t newTag)
...
+// be a different instance of that owner (e.g. a different motor channel) than the one configured.
+STATIC_UNIT_TESTED void printResourceClaimStatus(uint8_t resourceIndex, uint8_t index, ioTag_t ioTag) {
+    if (!ioTag) {
+        return;
+    }
+    const resourceOwner_e expectedOwner = resourceTable[resourceIndex].owner;
+    const resourceOwner_e actualOwner = timerGetOwner(ioTag);
+    // OWNER_FREE means this pin isn't under runtime timer-pinmap management, or the owning
+    // driver never called timerAllocate() this boot -- neither is a claim conflict to report.
+    if (actualOwner == OWNER_FREE || actualOwner == expectedOwner) {
```
[the remark this answers is not in the document]
Comment (nerdCopter) [reply src/main/interface/cli.c]: Fixed in 25faab776: added timerGetOwnerResourceIndex(), mirroring dmaopt's dmaGetOwner()/dmaGetResourceIndex() split. printResourceClaimStatus() now requires an exact RESOURCE_INDEX match for multi-instance owners (maxIndex > 0); single-instance owners (LED_STRIP, TRANSPONDER, CAMERA_CONTROL, BEEPER, PPMINPUT) pass a literal 0 resourceIndex into timerAllocate() and are unaffected. Added a regression test (SameOwnerDifferentIndexPrintsClaimedByWithIndex).

**Comment by nerdCopter** [comment]
@CodeRabbitAI approve
